renku/core/dataset/dataset_add.py

```diff
@@ -92,7 +93,9 @@ def add_to_dataset(
                     "Ignored adding paths under a .git directory:\n\t" + "\n\t".join(str(p) for p in paths_to_avoid)
                 )
 
-            files_to_commit = {f.get_absolute_commit_path(client.path) for f in files}
+            files_to_commit = {
+                f.get_absolute_commit_path(client.path) for f in files if not str(f.url).startswith("s3://")
```

Comment:
This is too specific to S3; we should make it a bit more general in case we add other external storages in future. Maybe by adding a hook function to `ProviderApi` to filter dataset files or some other approaches.

Reply:
I ended up adding `gitignored` field on the `DatasetAddMetadata` class. Is that an acceptable way to do this? It is `False` by default. And will be set to `True` for S3 and any other provider we have in the future can decide to do the same. Then this function becomes pretty general and understandable.

Reply:
Yes, that would work as well. You could also add an `IGNORE` action to `DatasetAddAction` which is included in `DatasetAddMetadata`.

Reply:
I will stick to the `gitignored` stuff I added. I think it is not impossible to use one of the existing actions and to have the file be "gitignored".